Source/src/WixSharp/Compiler.cs

```diff
@@ -3137,10 +3137,11 @@ static void ProcessCustomActions(Project wProject, XElement product)
                 else if (wAction is CustomActionRef)
                 {
                     var wCustomActionRef = (CustomActionRef) wAction;
+                    var whenstr = wCustomActionRef.When == When.After ? "After" : "Before";
                     sequences.ForEach(sequence =>
                         sequence.Add(new XElement("Custom", wCustomActionRef.Condition.ToXValue(),
                             new XAttribute("Action", wCustomActionRef.Id),
-                            new XAttribute("Before", wCustomActionRef.Step))));
+                            new XAttribute(whenstr, wCustomActionRef.Step))));
```

Comment:
You can just do this:
new XAttribute(wCustomActionRef.When.ToString(), wCustomActionRef.Step))));

Reply:
Indeed, it is simpler than my fix.